ospfd/ospf_interface.c

```diff
@@ -579,8 +579,7 @@ void ospf_free_if_params(struct interface *ifp, struct in_addr addr)
 	    && !OSPF_IF_PARAM_CONFIGURED(oip, type)
 	    && !OSPF_IF_PARAM_CONFIGURED(oip, auth_simple)
 	    && !OSPF_IF_PARAM_CONFIGURED(oip, auth_type)
-	    && listcount(oip->auth_crypt) == 0
-	    && ntohl(oip->network_lsa_seqnum) != OSPF_INITIAL_SEQUENCE_NUMBER) {
+	    && listcount(oip->auth_crypt) == 0) {
```

Comment:
Won't this change prevent the code from knowing the lsa sequence number of the interface when it goes up / down?  Shouldn't we be keeping track of that?  See

Reply:
See commit 7eb5b47e54169ac0da40ed1a6760db5d5f915a4d

Reply:
I saw this commit. First of all, there was a mistake here in the first place - we are removing the config when the sequence number IS NOT equal to the initial and we keep the config when the sequence number IS equal to the initial. So the idea of keeping learned sequence number didn't work anyway.

The sequence number is an operational variable and I don't think there is any point in storing it when the config is completely removed by the uses. More than that, it leads to a memory leak.

Reply:
I confirm that the sequence number is an operational variable computed when the LSA is build. When LSA is flush and then re-create (after removing and re-enabling configuration on the interface), the sequence number is re-computed. It is not a configured variable.